eng/testing/tests.props

```diff
@@ -32,6 +32,23 @@
     <SelfContained>true</SelfContained>
   </PropertyGroup>
 
+  <!-- when testing with Xharness we could host echo test server in the xharness process and avoid external server dependency -->
+  <PropertyGroup Condition="'$(TargetOS)' == 'Browser'">
+    <!-- send middleware to Helix and handle different path to it -->
+    <_TestEchoMiddleware Condition="'$(ContinuousIntegrationBuild)' == 'true' and '$(OS)' == 'Windows_NT'">%HELIX_CORRELATION_PAYLOAD%/xharness/TestEchoMiddleware</_TestEchoMiddleware>
+    <_TestEchoMiddleware Condition="'$(ContinuousIntegrationBuild)' == 'true' and '$(OS)' != 'Windows_NT'">$HELIX_CORRELATION_PAYLOAD/xharness/TestEchoMiddleware</_TestEchoMiddleware>
+    <_TestEchoMiddleware Condition="'$(ContinuousIntegrationBuild)' != 'true'">$(ArtifactsDir)bin/NetCoreServer/$(NetCoreAppCurrent)-$(Configuration)</_TestEchoMiddleware>
+
+    <WasmXHarnessArgs>$(WasmXHarnessArgs) --set-web-server-http-env=DOTNET_TEST_WEBSOCKETHOST,DOTNET_TEST_HTTPHOST,DOTNET_TEST_HTTP2HOST</WasmXHarnessArgs>
+    <WasmXHarnessArgs>$(WasmXHarnessArgs) --set-web-server-https-env=DOTNET_TEST_SECUREWEBSOCKETHOST,DOTNET_TEST_SECUREHTTPHOST,DOTNET_TEST_HTTP2SECUREHOST</WasmXHarnessArgs>
+    <WasmXHarnessArgs>$(WasmXHarnessArgs) --web-server-middleware=$(_TestEchoMiddleware)/NetCoreServer.dll,GenericHandler</WasmXHarnessArgs>
+  </PropertyGroup>
+
+  <ItemGroup Condition="'$(TargetOS)' == 'Browser'">
+    <ProjectReference Include="$(CommonTestPath)System/Net/Prerequisites/NetCoreServer/NetCoreServer.csproj" 
```

Comment:
Is this primarily here because of L40 that may point `_TestEchoMiddleware` at `NetCoreServer`? So this isn't used when its a `ContinuousIntegrationBuild`?

Reply:
This is here to make sure it's built. But none of the test use that dependency inside WASM process. It's needed both for CI and locally.